src/sidebar/util/groups.js

```diff
@@ -1,4 +1,25 @@
 import escapeStringRegexp from 'escape-string-regexp';
+import serviceConfig from '../service-config';
+
+/**
+ * Returns the value for allowLeavingGroups from the service config
+ * settings. By default this value is true. In that case, it is
+ * intended that groups can be left if they are private. If this
+ * value returns false, then the user can not leave any groups.
+ *
+ * @param {object} settings
+ * @return {boolean}
+ */
+function allowLeavingGroups(settings) {
```

Comment:
I think this might be simplified to something a little less verbose, e.g.:

```
function allowLeavingGroups(settings) {
  const config = serviceConfig(settings) || {};
  return typeof config.allowLeavingGroups === 'boolean'
    ? config.allowLeavingGroups
    : true;
}
```

Reply:
I'm fine with that, I more or less copied this from annotation-sharing.js so I'll go ahead and change both so we're at least being consitent.

Reply:
On second thought, nm, this is outside of scope.